include/BoostInstall.cmake

```diff
@@ -162,11 +162,11 @@ function(__boost_install_set_output_name LIB TYPE VERSION)
 
     else()
 
-      if(CMAKE_SYSTEM_PROCESSOR MATCHES "^[Aa][Rr][Mm]"
-           OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch")
-        set(arch "a")
-      elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^mips")
-        set(arch "m")
+      if(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|amd64)")
+        set(arch "x")
+      else()
+        string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
+        string(TOLOWER "${arch}" arch)
```

Comment:
```suggestion
      string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
      string(TOLOWER "${arch}" arch)
      if(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|amd64)")
        set(arch "x")
      endif()
```
Alternative?

Reply:
Could probably also move       
```
string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
string(TOLOWER "${arch}" arch)
```
out of the complete if block and remove the MSVC if block and merge the selection:
```
    string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
    string(TOLOWER "${arch}" arch)

    if(CMAKE_CXX_COMPILER_ARCHITECTURE_ID MATCHES "^ARM")
      set(arch "a")
    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|[aA][mM][Dd]64)"
        OR CMAKE_CXX_COMPILER_ARCHITECTURE_ID STREQUAL "X86"
        OR CMAKE_CXX_COMPILER_ARCHITECTURE_ID STREQUAL "x64"
      )
      set(arch "x")
    endif()
```